### ISSUE TITLE
IllegalStateException when calling "softmax" activation function on final layer of an RBM

### ISSUE THREAD

**Issue by KeironO** [deeplearning4j/deeplearning4j#538]
Pulled the recent version of dl4j/nd4j/canova, and I'm experiencing an IllegalStateException when setting the output layer to softmax.

```
Exception in thread "main" java.lang.IllegalStateException: Max along dimension for input must either be a row vector or scalar
    at org.nd4j.linalg.api.ops.impl.transforms.SoftMax.exec(SoftMax.java:179)
    at org.nd4j.linalg.api.ops.executioner.DefaultOpExecutioner.execAndReturnVector(DefaultOpExecutioner.java:501)
    at org.nd4j.linalg.api.ops.executioner.DefaultOpExecutioner.execAndReturn(DefaultOpExecutioner.java:486)
    at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.feedForwardActivationsAndDerivatives(MultiLayerNetwork.java:584)
    at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.doBackWard(MultiLayerNetwork.java:1079)
    at org.deeplearning4j.nn.multilayer.MultiLayerNetwork.fit(MultiLayerNetwork.java:1038)
    at org.deeplearning4j.Project.FloweringAndNotFloweringSmallScale.main(FloweringAndNotFloweringSmallScale.java:116)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:606)
    at com.intellij.rt.execution.application.AppMain.main(AppMain.java:140)
-- org.jblas INFO Deleting /tmp/jblas4353158275783117984/libjblas.so
-- org.jblas INFO Deleting /tmp/jblas4353158275783117984/libjblas_arch_flavor.so
-- org.jblas INFO Deleting /tmp/jblas4353158275783117984
```

**Comment by agibsonccc** [comment]
Mind submitting a test so we can reproduce this?

**Comment by KeironO** [comment]
Sure.

**Comment by KeironO** [comment]
(Apologies for not adding the link when I posted.)

**Comment by AlexDBlack** [comment]
Confirmed as also occurring for MLPBackpropIrisExample and BackPropMLPTest unit test. Seems to occur only for miniBatchSize > 1

**Comment by drlebedev** [comment]
It seems like the same issue: https://github.com/deeplearning4j/nd4j/issues/277 I faced for GravesLSTM. I did some debugging and it looks like maxAlongDimension should be either scalar or Column (not row) vector. Let me submit pull request in an hour.
